### PR TITLE
Added PIDCommand.py using ChatGPT translator

### PR DESCRIPTION
**Pull request by mbardoeChoate**
Trying to add in to the commands now. I used ChatGPT with the prompt in Chief Delphi. Let me know about what mistakes I might have made. I looked over some of the other commands to try to pick up style and imports. Thanks.

### CODE REVIEW

File: commands2/pidcommand.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        Creates a new PIDCommand, which controls the given output with a PIDController.
+
+        :param controller: the controller that controls the output.
+        :param measurementSource: the measurement of the process variable
+        :param setpoint: the controller's setpoint
+        :param useOutput: the controller's output
+        :param requirements: the subsystems required by this command
+        """
+        super().__init__()
+        if controller is None:
```
Comment (virtuald) [inline commands2/pidcommand.py:41]: It's a bit weird that you're checking all of these arguments to see if they're None. I know this is required in Java, but in python I think we should either do no check or check to see if the type is correct.

File: commands2/pidcommand.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        if setpoint is None:
+            raise ValueError("setpointSource must not be None")
+        if useOutput is None:
+            raise ValueError("useOutput must not be None")
+
+        self.controller = controller
+        self.useOutput = useOutput
+        self.measurement = measurementSource
+        self.setpoint = setpoint
+        self.requirements.addAll(set(requirements))
```
Comment (virtuald) [inline commands2/pidcommand.py:54]: Python's set has an `update` which is equivalent to `addAll`.

**Review by virtuald — COMMENTED** [review]
When adding stuff to this repo, it's really important to also include the unit tests from wpilibj at https://github.com/wpilibsuite/allwpilib/tree/main/wpilibNewCommands/src/test/java/edu/wpi/first/wpilibj2/command ... which don't exist for this command, never mind then (but keep it in mind for the future)

**Comment by virtuald** [comment]
If we're going to lean on ChatGPT for doing this, we should probably do https://github.com/robotpy/robotpy-commands-v2/issues/29 first to make sure we catch some of the more common errors.

File: commands2/pidcommand.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    are performed synchronously in the command's execute() method.
+
+    This class is provided by the NewCommands VendorDep
+    """
+
+    def __init__(
+        self,
+        controller: PIDController,
+        measurementSource: Callable[[], float],
+        setpoint: float,
```
[the remark this answers is not in the document]
Comment (mbardoeChoate) [reply commands2/pidcommand.py]: This makes sense to me, but I was wondering if there is a source or example you would recommend for how to deal with other instances where there are multiple constructors, and how to best handle those.

[the remark this answers is not in the document]
Comment (virtuald) [reply commands2/pidcommand.py]: The only thing I would suggest is either other commands, or see robotpy-wpilib prior to 2020 (which was all done in pure python). For example, here's the first command implementation from 2019: https://github.com/robotpy/robotpy-wpilib/tree/2019/wpilib/wpilib/command

There is also https://robotpy.readthedocs.io/en/2019.1.2/dev/porting.html#syntax-style-guide but it doesn't address this specifically.

[the remark this answers is not in the document]
Comment (virtuald) [reply commands2/pidcommand.py]: In general though we want to support the same functionality as the Java library even if the semantics differ slightly, so at the end of the day it's often a judgement call. Do what makes sense, and we can argue about it. :)

**Comment by virtuald** [comment]
Thanks!

**Pull request merged**